src/ssl.c

```diff
@@ -34566,15 +34570,17 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
 
         /* copy contents */
         XMEMCPY(dup, name, sizeof(WOLFSSL_X509_NAME));
+        InitX509Name(dup, 1);
+        /* Need to set dynamicName before copying */
+        dup->dynamicName = 1;
+        dup->sz = name->sz;
 
         /* handle dynamic portions */
-        if (name->dynamicName) {
-            if (!(dup->name = (char*)XMALLOC(name->sz, 0,
-                                             DYNAMIC_TYPE_OPENSSL))) {
-                goto err;
-            }
-            XMEMCPY(dup->name, name->name, name->sz);
+        if (!(dup->name = (char*)XMALLOC(name->sz, 0,
```

Comment:
dup->name should only be dynamic if name->name is dynamic, right?  Otherwise dup->name already points to dup->staticName, with dup->dynamicName = 0, and we can just XMEMCPY name->name into dup->name.

In this case, 4 lines above, "dup->dynamicName" would still be 0 if static.

Reply:
Fixed in latest commit